test/lib/Benchmark.js

```diff
@@ -0,0 +1,259 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+// The default benchmark time is 1 second, i.e. 1e6 microseconds
+const DEFAULT_BENCH_TIME = 1e6;
+
+// For the performance.now(), so the timer resolution is 1 microsecond
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+// The max iteration number is 1e6 as the timer resolution is 1 microsecond
+const MAX_ITERATION = 1e6;
+
+/*
+ * Get the DOMHighResTimeStamp in microsecond
+ * See more detail at https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
+ */
+function _performanceNow() {
+  return performance.now() * 1e3;
+}
+
+/*
+ * A simple benchmark framework for Activity Stream that runs on top of jpm test
+ *
+ * If you're new to Javascript benchmarking, here is a list of references:
+ *
+ *   https://developer.mozilla.org/en-US/docs/Mozilla/Benchmarking
+ *   https://mathiasbynens.be/notes/javascript-benchmarking
+ *   http://ejohn.org/blog/javascript-benchmark-quality/
+ *
+ * Suitable use cases:
+ *   * measure the load latency of Activity Stream with different profiles and browsering history
```

Comment:
I agreed with you that the current methodology won't work in certain cases. It hasn't yet done too much to create a reasonably fair benchmark environment. Like, force a GC before each benchmark, turn off the JIT optimization, or other system level tweaks. I am open to any suggestions on that though. 

My understanding of this benchmark thing is,
* try the best first, having a working one is better than flying in blind, although it could start from being inaccurate in some cases. We can improve it as we understand more in an incremental fashion
* the benchmark designer also could take some actions inside the benchmark function to mitigate the side-effect. i.e. clean up the cache, reset the table in the database etc

Let's have a chat about this to sort things out, Dan. Really appreciate your insights!